client/src/extension.ts

```diff
@@ -1,43 +1,50 @@
-import * as path from 'path';
-import { workspace, ExtensionContext, commands, languages, Disposable } from 'vscode';
+import { join } from 'path';
+import { workspace, ExtensionContext, languages, Disposable } from 'vscode';
 
 import {
     LanguageClient,
     LanguageClientOptions,
     ServerOptions,
-    TransportKind,
+    TransportKind
 } from 'vscode-languageclient';
 import OMTLinkProvider from './omtLinkProvider';
 
 let client: LanguageClient;
 
 export function activate(context: ExtensionContext) {
     // The server is implemented in node
-    let serverModule = context.asAbsolutePath(
-        path.join('server', 'out', 'server.js')
+    const serverModule = context.asAbsolutePath(
+        join('server', 'out', 'server.js')
     );
     // The debug options for the server
     // --inspect=6009: runs the server in Node's Inspector mode so VS Code can attach to the server for debugging
-    let debugOptions = { execArgv: ['--nolazy', '--inspect=6009'] };
+    const debugOptions = { execArgv: ['--nolazy', '--inspect=6009'] };
 
     // If the extension is launched in debug mode then the debug server options are used
     // Otherwise the run options are used
-    let serverOptions: ServerOptions = {
+    const serverOptions: ServerOptions = {
         run: { module: serverModule, transport: TransportKind.ipc },
         debug: {
             module: serverModule,
             transport: TransportKind.ipc,
             options: debugOptions
-        }
+        },
     };
 
     // Options to control the language client
-    let clientOptions: LanguageClientOptions = {
+    const clientOptions: LanguageClientOptions = {
         // Register the server for OMT and ODT documents
         documentSelector: [
             { scheme: 'file', language: 'omt' },
             { scheme: 'file', language: 'odt' },
-        ]
+        ],
+        synchronize: {
+            // Notify the server about changes to files conforming to these patterns:
+            fileEvents: [
+                workspace.createFileSystemWatcher('**/*.omt'),
+                workspace.createFileSystemWatcher('**/package.json'),
```

Comment:
Why does the language server need to know about the `package.json`s?

Reply:
it uses the paths defined in them to resolve shorthand uri imports.

Reply:
Aren't those defined in the `tsconfig.json`s?

Reply:
getting frustrated with myself, not registering that. You are right ofcourse.. changed it.